### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by MKumaran98**

### CODE REVIEW

**Review by DanishOnGit — COMMENTED** [review]
LGTM !

File: Config/Passport.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const opts={
+    jwtFromRequest : ExtractJwt.fromAuthHeaderAsBearerToken(),
+    secretOrKey :process.env["SECRET"]
+}
+
+passport.use(new JwtStrategy(opts, function(jwt_payload, done) {
+    usersdb.findById(jwt_payload._id, function(err, user) {
+        if (err) {
+            return done(err, false);    //error
```
Comment (DarkShadow20) [inline Config/Passport.js:14]: you should remove the comments

File: Controller/CartWishlist.controller.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    const {id}=req.params;
+    const {productId}=req.body;
+    try{
+        const user=await usersdb.findById(id);
+        const cart=await cartsdb.findById(user.cart);
+        if(cart){
+            if(!cart.cartItems.some(({product})=>product==productId)){
+                cart.cartItems.push({
+                    product:productId,
+                    quantity:1
```
Comment (DarkShadow20) [inline Controller/CartWishlist.controller.js:16]: you should not hardcode the value of quantity

File: Controller/CartWishlist.controller.js [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+module.exports.addToWishlist=async (req,res)=>{
+    const {id}=req.params;
+    const {productId}=req.body;
+    try{
+        const user=await usersdb.findById(id);
+        const wishlist=await wishlistsdb.findById(user.wishlist);
+        if(wishlist){
+            if(!wishlist.wishlistItems.some(product=>product==productId)){
```
Comment (DarkShadow20) [inline Controller/CartWishlist.controller.js:146]: good practice to use === instead of ==

File: Controller/CartWishlist.controller.js [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    const {id,productId}=req.params
+    try{
+        const user=await usersdb.findById(id)
+        const wishlist=await wishlistsdb.findById(user.wishlist);
+        if(wishlist.wishlistItems.some(product=>product==productId)){
+            await wishlist.update({$pull:{wishlistItems:productId}})
+        }else{
+            return res.status(400).json({
+                ok:false,
+                messafe:"Invalid request"
```
Comment (DarkShadow20) [inline Controller/CartWishlist.controller.js:187]: message spelling is not correct

File: Controller/CartWishlist.controller.js (hunk 2 above)
Comment (MKumaran98) [reply Controller/CartWishlist.controller.js:16]: Hey Kunal I'll refactor it!

File: Controller/CartWishlist.controller.js (hunk 4 above)
Comment (MKumaran98) [reply Controller/CartWishlist.controller.js:187]: Good catch! I'll change it

**Pull request merged**